util/deephash/types_test.go

```diff
@@ -12,8 +12,19 @@ import (
 
 	"tailscale.com/tailcfg"
 	"tailscale.com/types/structs"
+	"tailscale.com/util/hashx"
 )
 
+type implsSelfHasher struct {
+	emit uint64
+	// references self so typeIsRecursive would erroneously return true
```

Comment:
why's that important?

Reply:
I guess its less important now i have good coverage through TestHash